reactor-netty-http/src/main/java/reactor/netty/http/client/Http2Pool.java

```diff
@@ -140,23 +149,33 @@ final class Http2Pool implements InstrumentedPool<Connection>, InstrumentedPool.
 			AtomicIntegerFieldUpdater.newUpdater(Http2Pool.class, "wip");
 
 	final Clock clock;
+	final Long maxConcurrentStreams;
+	final long maxIdleTime;
 	final long maxLifeTime;
+	final int minConnections;
 	final PoolConfig<Connection> poolConfig;
+	final LongAdder totalMaxConcurrentStreams = new LongAdder();
 
```

Comment:
The totalMaxConcurrentStreams is a LongAdder but it seems to be less frequently updated than read ?
is it correct ? for example, the totalMaxConcurrentStreams is only updated when a slot is created or when the server has sent a SETTINGS with an updated max-concurrent-stream, in this case, the canOpenStream() method will update the totalMaxConcurrentStreams (but only if the diff is != 0):

```
		boolean canOpenStream() {
			ChannelHandlerContext frameCodec = http2FrameCodecCtx();
			if (frameCodec != null && http2MultiplexHandlerCtx() != null) {
				long maxActiveStreams = ((Http2FrameCodec) frameCodec.handler()).connection().local().maxActiveStreams();
				maxActiveStreams = pool.maxConcurrentStreams == -1 ? maxActiveStreams :
						Math.min(pool.maxConcurrentStreams, maxActiveStreams);
				long diff = maxActiveStreams - maxConcurrentStreams;
->				if (diff != 0) {
					maxConcurrentStreams = maxActiveStreams;
					pool.totalMaxConcurrentStreams.addAndGet(diff);
				}
				int concurrency = this.concurrency;
				return concurrency < maxActiveStreams;
			}
			return false;
		}
```
However, the totalMaxConcurrentStreams is read very very frequently by the request() method.

So, what I mean is that even if under low update contentions a LongAdder performance is similar to an AtomicInteger, it however has higher space consumption and is less performant than AtomicInteger in terms of read access.

Then why not using an AtomicInteger instead of a LongAdder for the totalMaxConcurrentStreams ?

Reply:
fixed